### ISSUE TITLE
index out of range on get_deployment_json

### ISSUE THREAD

**Issue by AliKarami** [telepresenceio/telepresence#949]
### What were you trying to do?

Simply run the telepresence.

### What did you expect to happen?

simply run!

### What happened instead?

it creates the telepresence deployment in my OKD project, but suddenly an error popped out: `IndexError: list index out of range`

### Automatically included information

Command line: `['/usr/local/bin/telepresence']`
Version: `0.97`
Python version: `3.6.5 |Anaconda, Inc.| (default, Apr 26 2018, 08:42:37) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]`
kubectl version: `Client Version: v1.10.11 // Server Version: v1.10.0+b81c8f8`
oc version: `oc v3.11.0+0cbc58b // kubernetes v1.11.0+d4cacc0 // features: Basic-Auth //  // Server https://****************************:443 // openshift v3.10.0+91d44f1-52 // kubernetes v1.10.0+b81c8f8`
OS: `Darwin AliKBookPro.local 18.2.0 Darwin Kernel Version 18.2.0: Thu Dec 20 20:46:53 PST 2018; root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64`

```
Traceback (most recent call last):
  File "/usr/local/bin/telepresence/telepresence/cli.py", line 130, in crash_reporting
    yield
  File "/usr/local/bin/telepresence/telepresence/main.py", line 60, in main
    remote_info = start_proxy(runner)
  File "/usr/local/bin/telepresence/telepresence/proxy/__init__.py", line 104, in start_proxy
    run_id=run_id,
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 156, in get_remote_info
    runner, deployment_name, deployment_type, run_id=run_id
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 103, in get_deployment_json
    )["items"][0]
IndexError: list index out of range

```

Logs:

```
ging delete --ignore-not-found svc,deploy --selector=telepresence=537d1ea1913047e1a05be107b7586af2
   5.5 TEL | [11] Capturing: oc --context my-staging/*****************:443/AliKarami --namespace my-staging run --restart=Always --limits=cpu=100m,memory=256Mi --requests=cpu=25m,memory=64Mi telepresence-1552458933-9747179-39452 --image=datawire/telepresence-k8s:0.97 --labels=telepresence=537d1ea1913047e1a05be107b7586af2
   6.2  11 | deploymentconfig.apps.openshift.io/telepresence-1552458933-9747179-39452 created
   6.2 TEL | END SPAN deployment.py:61(create_new_deployment)    1.5s
   6.2 TEL | BEGIN SPAN remote.py:154(get_remote_info)
   6.2 TEL | BEGIN SPAN remote.py:78(get_deployment_json)
   6.2 TEL | [12] Capturing: oc --context my-staging/****************:443/AliKarami --namespace my-staging get deployment -o json --export --selector=telepresence=537d1ea1913047e1a05be107b7586af2
   7.0 TEL | END SPAN remote.py:78(get_deployment_json)    0.8s

```

**Comment by jdelafon** [comment]
The real problem is the non-meaningful error message, should be "No deployment was found" instead of IndexError.

**Comment by upendran** [comment]
For me the issue was due to kubectl version. I had to downgrade to v1.15.1 to fix this issue

**Comment by tobisinghania** [comment]
> For me the issue was due to kubectl version. I had to downgrade to v1.15.1 to fix this issue

For me downgrading was also the only way to get telepresence to work again. Unfortunately, in the long run using an old version of kubectl cannot be an option.

**Comment by Riimu** [comment]
I also ran into this issue. For me, the problem might be due to the kubernetes-cli version 1.18.2 conflicting somehow with our cluster version 1.15.10.

Temporary solution for me was to install previous version via homebrew using:

```
brew unlink kubernetes-cli
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/7104022700e3c6eff3ea5b3f61f52fb9ebc3800c/Formula/kubernetes-cli.rb
```

The git commit points to 1.17.4 version of the kubernetes-cli.

**Comment by ark3** [comment]
I believe this was fixed by #1404 for OpenShift users (to be released shortly) and in release 0.105 for users of newer `kubectl`.
